axelrod/tests/unit/test_pickling.py

```diff
@@ -0,0 +1,335 @@
+    st.DualTransformer(),
+    st.FlipTransformer(),
+    st.FinalTransformer((D, D)),
+    st.ForgiverTransformer(0.2),
+    st.GrudgeTransformer(3),
+    st.InitialTransformer((C, D)),
+    st.JossAnnTransformer((0.2, 0.6)),
+    st.MixedTransformer(probability, strategies),
+    st.NiceTransformer(),
+    st.NoisyTransformer(0.2),
+    st.RetaliationTransformer(3),
+    st.RetaliateUntilApologyTransformer(),
+    st.TrackHistoryTransformer(),
+    st.IdentityTransformer()
+]
+
+
+class TestPickle(unittest.TestCase):
+
+    def assert_orignal_equals_pickled(self, player, turns=10):
+        self.assert_original_plays_same_as_pickled(player, turns)
+        self.assert_mutated_instance_same_as_pickled(player)
+
+    def assert_original_plays_same_as_pickled(self, player, turns=10):
+        copy = pickle.loads(pickle.dumps(player))
+        opponent_1 = axl.CyclerCCCDCD()
+        opponent_2 = axl.CyclerCCCDCD()
+        axl.seed(0)
+        match_1 = axl.Match((player, opponent_1), turns=turns)
+        result_1 = match_1.play()
+
+        axl.seed(0)
+        match_2 = axl.Match((copy, opponent_2), turns=turns)
+        result_2 = match_2.play()
+
+        self.assertEqual(result_1, result_2)
+
+    def assert_mutated_instance_same_as_pickled(self, player):
+        player.reset()
+        turns = 5
+        opponent = axl.Alternator()
+        for _ in range(turns):
+            player.play(opponent)
+        new = pickle.loads(pickle.dumps(player))
+        self.assertEqual(player, new)
+
+    def test_parameterized_player(self):
+        player = axl.Cycler('DDCCDD')
+        self.assert_orignal_equals_pickled(player)
+
+    def test_sequence_player(self):
+        player = axl.ThueMorse()
+        self.assert_orignal_equals_pickled(player)
+
+    def test_final_transformer_called(self):
+        player = axl.Alexei()
+        copy = pickle.loads(pickle.dumps(player))
+        match = axl.Match((player, copy), turns=3)
+        results = match.play()
+        self.assertEqual(results, [(C, C), (C, C), (D, D)])
+
+        self.assert_original_plays_same_as_pickled(axl.Alexei(), turns=10)
+
+    def test_nice_transformer_class(self):
+        player = axl.NMWEDeterministic()
+        self.assert_original_plays_same_as_pickled(player, turns=10)
+
+    def test_pickling_all_strategies(self):
+        for s in axl.strategies:
+            self.assert_orignal_equals_pickled(s())
+
+    def test_pickling_all_transformers_as_decorated_classes(self):
+        for s in transformed_no_prefix:
+            player = s()
+            self.assert_orignal_equals_pickled(player)
+
+    def test_pickling_all_transformers_as_instance_called_on_a_class(self):
+        for transformer in transformer_instances:
+            player = transformer(axl.Cooperator)()
+            self.assert_orignal_equals_pickled(player)
+
+    def test_created_on_the_spot_multiple_transformers(self):
+        klass = st.FlipTransformer()(axl.Cooperator)
```

Comment:
i picked "klass" as it seems to be convention in **a lot** of code.  i also have no real preference.  

then i search on PEP8 and here's what i found.

>If a function argument's name clashes with a reserved keyword, it is generally better to append a single trailing underscore rather than use an abbreviation or spelling corruption. Thus class_ is better than clss . (Perhaps better is to avoid such clashes by using a synonym.)

sooo... `class_` seems official. i'll go with that.